### PR TITLE
Add remaining infrastructure tests for empty selections

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
Part of overall project: https://github.com/tidymodels/recipes/issues/1130
dealing with part of https://github.com/tidymodels/recipes/issues/1134 by adding tests for

- Empty printing
- printing
- empty selection prep/bake is a no-op

Doing the other 2 types in a different PR for ease of review.

### CODE REVIEW

File: NEWS.md
Comment (hfrick) [inline NEWS.md:1]: Given that different steps did different things for "empty selection", it'd be nice to add what they do now (which is "nothing").

File: tests/testthat/_snaps/geodist.md [hunk 1]
```diff
@@ -162,6 +162,41 @@
       Error in `step_geodist()`:
       ! `is_lat_lon` should be a single logical value.
 
+# empty printing
+
+    Code
+      rec
```
Comment (hfrick) [inline tests/testthat/_snaps/geodist.md:168]: What do you think about adding the definition of `rec` to the snapshots? Then we could see that it's testing the right thing without referring to the `test-*.R` files.

Comment (EmilHvitfeldt) [reply tests/testthat/_snaps/geodist.md:168]: it is not a bad idea. 

I think I like to stay with the current design, as all the steps follow this formula with a couple of small variations.

```r
test_that("empty printing", {
  rec <- recipe(mpg ~ ., mtcars)
  rec <- name_of_step_here(rec)

  expect_snapshot(rec)

  rec <- prep(rec, mtcars)

  expect_snapshot(rec)
})

```
